File: YZTemplate/YZTableViewController.xctemplate/UIViewControllerXIBObjective-C/___FILEBASENAME___.m
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

#import "___FILEBASENAME___.h"

#pragma mark - define

@interface ___FILEBASENAMEASIDENTIFIER___ ()<UITableViewDataSource, UITableViewDelegate>

@property (nonatomic, weak) IBOutlet UITableView *tableView;
@property (nonatomic, strong) NSMutableArray *dataSource;

@end

@implementation ___FILEBASENAMEASIDENTIFIER___

#pragma mark - Initial Methods

#pragma mark - Override Methods
- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self setupUIComponent];
}

/*
- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
}

- (void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
}

-(void)viewDidDisappear:(BOOL)animated {
    [super viewDidDisappear:animated];
}
**/

#pragma mark - Setup UI Component
- (void)setupUIComponent {
    [self setupTableView];
}

- (void)setupTableView {
    
}

#pragma mark - Public Methods

#pragma mark - View Event

#pragma mark - Target Action

#pragma mark - Private Method

#pragma mark - External Delegate

#pragma mark - UITableViewDataSource
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.dataSource.count;
}

static NSString * const kCellIdentifier = @"";
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:kCellIdentifier forIndexPath:indexPath];
    
    return cell;
}

#pragma mark - UITableViewDelegate
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    
}

#pragma mark - Notification

#pragma mark - Getter Methods
- (NSMutableArray *)dataSource {
    if (!_dataSource) {
        _dataSource = [NSMutableArray array];
    }
    return _dataSource;
}

#pragma mark - Setter Methods

@end

File: YZTemplate/YZTableViewController.xctemplate/UIViewControllerSwift/___FILEBASENAME___.swift
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

import UIKit

class ___FILEBASENAMEASIDENTIFIER___: UIViewController {
    
    @IBOutlet internal var tableView: UITableView!
    var dataSource = [AnyObject]()
    
    //MARK: - Intial Methods
    
    //MARK: - Override Methods
    override func viewDidLoad() {
        super.viewDidLoad()
        
        setupUIComponent()
    }
    
    //MARK: - Setup UI Component
    func setupUIComponent() {
        setupTableView()
    }
    
    func setupTableView() {
        
    }
    
    //Mark: - Public Methods
    
    //Mark: - View Event
    
    //Mark: - Target Action
    
    //Mark: - Private Method
    
    //Mark: - External Delegate
    
    //Mark: - Notification
    
    //Mark: - Getter Methods
    
    //Mark: - Setter Methods
}

extension ___FILEBASENAMEASIDENTIFIER___: UItableViewDataSource {
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return dataSource.count
    }
    
    private static let kCellIndentifier = ""
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: ___FILEBASENAMEASIDENTIFIER___.kCellIndentifier, for: indexPath)

        return cell
    }
}

extension ___FILEBASENAMEASIDENTIFIER___: UItableViewDelegate {
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        
    }
}
